Whenever a new ML algorithm is created, creators often select one or more models as based model for comparision purpose. For that reason, this tutorial is .

In [18]:
from sklearn import dummy
from sklearn.metrics import accuracy_score
from sklearn.datasets import make_regression, make_classification

In [2]:
X,y = make_regression()
dumdum = dummy.DummyRegressor()
dumdum.fit(X,y)

DummyRegressor(constant=None, quantile=None, strategy='mean')

## Strategies:
There are several approaches to predict with this dummy model:   
    1. Mean    
    2. Median   
    3. Constant   

In [7]:
predictors =[('mean', None),('median',None),('constant',10)]
for strategy, constant in predictors:
    dumdum = dummy.DummyRegressor(strategy=strategy, constant=constant)
    dumdum.fit(X,y)
    print("strategy {}".format(strategy),",".join(map(str,dumdum.predict(X)[:5])) )

strategy mean 8.8653720943,8.8653720943,8.8653720943,8.8653720943,8.8653720943
strategy median -18.4574371803,-18.4574371803,-18.4574371803,-18.4574371803,-18.4574371803
strategy constant 10.0,10.0,10.0,10.0,10.0


## For classification, we can

In [15]:
predictors = [("constant",0),("stratified",None),("uniform",None),("most_frequent",None)]
X,y = make_classification()
for strategy, constant in predictors:
    dumdum = dummy.DummyClassifier(strategy=strategy, constant=constant)
    dumdum.fit(X,y)
    print("strategy: {}".format(strategy),",".join(map(str, dumdum.predict(X)[:5])))

strategy: constant 0,0,0,0,0
strategy: stratified 1,1,0,0,1
strategy: uniform 1,0,1,1,0
strategy: most_frequent 0,0,0,0,0


## Apply
Let say we want to test a fraud detection model such that only 5% of dataset is fraudulent. We may think to fit model by never do anything.

In [17]:
X,y = make_classification(20000, weights=[0.95,.05])
dumdum = dummy.DummyClassifier(strategy='most_frequent')
dumdum.fit(X,y)

DummyClassifier(constant=None, random_state=None, strategy='most_frequent')

In [19]:
print(accuracy_score(y, dumdum.predict(X)))

0.9455


## Well, our baseline model achieve 94.5% 
so any new model generates less accurate than that, it is worthless